### ISSUE TITLE
md5 as file name

### ISSUE THREAD

**Issue by AdClear247** [qsniyg/maxurl#748]
Is it possible to make file name to be md5 checksum of file? I frequently find some pictures damaged on my truecrypt drive and I would like to know if they are damaged by md5 even if not visually different. Although for more photos using such tools as Total Commander, HashMyFiles, HashCheck Shell Extension would be easier because I could hash all files in one folder and verify by simple double click. Also md5 of url would be another good option at least because of uniqueness and allowed characters in filename.

**Comment by qsniyg** [comment]
Good idea! I'll have to rewrite the way it's handled though, which is also the blocker for #721.

MD5 checksum of the url is easy, the file itself is going to be a bit harder though, especially since partial images aren't actually loaded by the script (meaning, if partial media is enabled, then it'll have to request it twice, once by the browser, once by the script).
